packages/atomic/src/components/commerce/atomic-product/atomic-product.ts

```diff
@@ -250,16 +254,8 @@ export class AtomicProduct extends ChildrenUpdateCompleteMixin(LitElement) {
     return parentNodeToString(this.linkContent ?? new HTMLElement());
   }
 
-  private shouldExecuteRenderFunction() {
-    return (
-      this.isCustomRenderFunctionMode &&
-      this.productRootRef &&
-      !this.executedRenderingFunctionOnce
-    );
-  }
-
   public render() {
-    if (this.isCustomRenderFunctionMode) {
+    if (this.customRenderController.hasCustomRenderFunction) {
```

Comment:
Honestly the more I look at it, the more I think we should just get rid of that public method on the reactive controller! It's such a trivial check.